Content.Client/GameObjects/Components/Mobs/ClientStatusEffectsComponent.cs

```diff
@@ -0,0 +1,139 @@
+        private readonly IDictionary<StatusEffect, TextureRect> _icons = new Dictionary<StatusEffect, TextureRect>();
+
+        /// <summary>
+        /// Allows calculating if we need to act due to this component being controlled by the current mob
+        /// </summary>
+        private bool CurrentlyControlled => _playerManager.LocalPlayer.ControlledEntity == Owner;
+
+        public override void OnAdd()
+        {
+            base.OnAdd();
+            PlayerAttached();
+        }
+
+        protected override void Shutdown()
+        {
+            base.Shutdown();
+            PlayerDetached();
+        }
+
+        public override void HandleMessage(ComponentMessage message, INetChannel netChannel = null,
+            IComponent component = null)
+        {
+            base.HandleMessage(message, netChannel, component);
+            switch (message)
+            {
+                case StatusEffectsMessage msg:
+                    if (!CurrentlyControlled)
+                    {
+                        break;
+                    }
+                    switch (msg.Mode)
+                    {
+                        case StatusEffectsMode.Change:
+                            ChangeIcon(msg.Name, msg.Filepath);
+                            break;
+                        case StatusEffectsMode.Remove:
+                            RemoveIcon(msg.Name);
+                            break;
+
+                        default:
+                            throw new ArgumentOutOfRangeException();
+                    }
+                    break;
+                case PlayerAttachedMsg _:
+                    if (!CurrentlyControlled)
+                    {
+                        break;
+                    }
+                    PlayerAttached();
+                    break;
+                case PlayerDetachedMsg _:
+                    if (!CurrentlyControlled)
+                    {
+                        break;
+                    }
+                    PlayerDetached();
+                    break;
+            }
+        }
+
+        private void PlayerAttached()
+        {
+            if (_ui != null)
+            {
+                return;
+            }
+            _ui = new StatusEffectsUI();
+            _userInterfaceManager.StateRoot.AddChild(_ui);
+        }
+
+        private void PlayerDetached()
+        {
+            _ui?.Dispose();
+            _icons.Clear();
+        }
+
+        public void ChangeIcon(StatusEffect name, string filepath)
+        {
+            // Create
+            if (!_icons.ContainsKey(name))
+            {
+                TextureRect newIcon = new TextureRect
+                {
+                    TextureScale = (2, 2),
+                    Texture = _resourceCache.GetTexture(filepath)
+                };
+                _icons.Add(name, newIcon);
+                // TODO: May be a better way to do this but I don't imagine they would update that frequently
```

Comment:
If you really care you could go and do `SetPositionInParent` combined with a binary search or whatever to insert thee efficiently.

But yeah, I really doubt it's gonna be a problem in the foreseeable future.